File: Excel/900-999/949/949 Challenge.py
import pandas as pd
import numpy as np

path = "900-999/949/949 Data Transpose.xlsx"
input = pd.read_excel(path, usecols="A:C", skiprows=1, nrows=23)
test  = pd.read_excel(path, usecols="E:G", skiprows=1, nrows=15)

m = input.to_numpy(dtype=object)
m = m[~np.all(pd.isna(m), axis=1)]
is_date = pd.Series(m[:, 0]).astype(str).str.match(r"^\d")
shift   = (~is_date) & pd.isna(m[:, 2])
m[shift] = np.column_stack([
    [np.nan] * shift.sum(),
    m[shift, 0],
    m[shift, 1]
])

result = pd.DataFrame(m, columns=input.columns)
result['Col1'] = result['Col1'].ffill()
result = result[~result['Col2'].isna()].reset_index(drop=True)
result['Col3'] = result['Col3'].astype("int64")
result.columns = test.columns


print(result.equals(test))
# True

File: Excel/900-999/949/949 Challenge.R
library(tidyverse)
library(readxl)
library(janitor)

path <- "900-999/949/949 Data Transpose.xlsx"
input <- read_excel(path, range = "A2:C24")
test <- read_excel(path, range = "E2:G17")

m <- as.matrix(input)
m <- m[rowSums(is.na(m)) < ncol(m), ]
is_date <- grepl("^\\d", m[, 1])
shift <- !is_date & is.na(m[, 3])
m[shift, ] <- cbind(NA, m[shift, 1], m[shift, 2])

result = as_tibble(m) %>%
  fill(Col1, .direction = "down") %>%
  na.omit() %>%
  mutate(
    Col1 = excel_numeric_to_date(as.numeric(Col1)) %>% as.POSIXct(),
    Col3 = as.numeric(Col3)
  )

all.equal(result, test, check.attributes = FALSE)
